bin/ns

```diff
@@ -4,91 +4,29 @@
 require_relative '../lib/auto_hck'
 
 module AutoHCK
-  # rubocop:disable Metrics/BlockLength
   run do
-    # rubocop:enable Metrics/BlockLength
-
-    require 'English'
-    require 'etc'
-    require 'fileutils'
-    require 'tmpdir'
-
-    euid = Process.euid
-    if euid != 0
-      pwname = Etc.getpwuid(Process.uid).name
-
-      # getsubids is not available on some supported systems.
-      subid_start, subid_count = File.open('/etc/subgid') do |file|
-        begin
-          file.readline while file.readline(':', chomp: true) != pwname
-        rescue EOFError
-          abort <<~MESSAGE
-            subgid delegation is not set up for the current user.
-            See newgidmap(1) for details.
-          MESSAGE
-        end
-
-        [file.readline(':', chomp: true), file.readline(chomp: true)]
-      end
+    require 'optparse'
+
+    chdir = Dir.pwd
+    Dir.chdir(File.dirname(__dir__))
+
+    workspace_path = nil
+    parser = OptionParser.new
+    parser.banner += ' [command]'
+    parser.version = VERSION
+    parser.on('-w <path>', '--workspace <path>', String, 'Workspace path') { workspace_path = _1 }
+    argv = parser.order(ARGV)
+
+    if workspace_path.nil?
+      Dir.chdir Config.read['workspace_path']
+      dirs = Dir['*/*/*/']
+      abort 'no workspace found' if dirs.empty?
+
+      workspace_path = dirs.max_by { File.mtime(_1) }
```

Comment:
This can be broken when several instances of AutoHCK run in parallel. I think we should fail without a workspace.

Reply:
It is a convenient fallback for development. You usually don't manually invoke `bin/ns`.